sql/handler.h

```diff
@@ -1713,59 +1713,64 @@ struct Vers_parse_info
   start_end_t system_time;
   start_end_t as_row;
 
-  void set_period_for_system_time(LString start, LString end)
+  void set_system_time(LString start, LString end)
   {
     system_time.start= start;
     system_time.end= end;
   }
 
-private:
-  bool is_trx_start(const char *name) const;
-  bool is_trx_end(const char *name) const;
-  bool is_trx_start(const Create_field &f) const;
-  bool is_trx_end(const Create_field &f) const;
-  bool fix_implicit(THD *thd, Alter_info *alter_info, bool integer_fields);
+protected:
+  friend struct Table_scope_and_contents_source_st;
+  void set_start(const LEX_CSTRING field_name)
+  {
+    as_row.start= field_name;
+    system_time.start= field_name;
+  }
+  void set_end(const LEX_CSTRING field_name)
+  {
+    as_row.end= field_name;
+    system_time.end= field_name;
+  }
+  bool is_start(const char *name) const;
+  bool is_end(const char *name) const;
+  bool is_start(const Create_field &f) const;
+  bool is_end(const Create_field &f) const;
+  bool fix_implicit(THD *thd, Alter_info *alter_info, bool integer_fields, int *added= NULL);
+  bool any_sys_field_declared() const
```

Comment:
Name can be better. It's not only check for sys fields, but also `PERIOD FOR SYSTEM_TIME` parts.